ESF-905 Trustly documentation (#421)
SwedbankPay/developer.swedbankpay.com: payments/trustly/seamless-view.md

---
title: Swedbank Pay Trustly Payments
sidebar:
  navigation:
  - title: Trustly Payments
    items:
    - url: /payments/trustly
      title: Introduction
    - url: /payments/trustly/redirect
      title: Redirect
    - url: /payments/trustly/seamless-view
      title: Seamless View
    - url: /payments/trustly/after-payment
      title: After Payment
    - url: /payments/trustly/other-features
      title: Other Features
---

{% include jumbotron.html body="The Seamless View scenario represents the
                                opportunity to implement Trustly directly in
                                your webshop." %}

## Introduction

Seamless View provides an integration of the payment process directly on your
website. This solution offers a smooth shopping experience with Swedbank Pay

> payment pages seamlessly integrated in an `iframe` on your website. The costumer

Reviewer: ```suggestion
Payments seamlessly integrated in an `iframe` on your website. The payer
```

does not need to leave your webpage, since we are handling the payment in the

> `iframe` on your page.

Reviewer: ```suggestion
`iframe` on your website.
```

![screenshot of the Trustly payment window][trustly-payment-embedded-view]{:height="425px" width="700px"}

{% include alert-callback-url.md api_resource="trustly" %}

## Step 1: Create the payment

> A Trustly payment is a straightforward way to perform a direct-bank payment. It is followed up by posting a capture, cancellation or reversal

Reviewer: It was previously mentioned that only `intent: Sales` is supported. If that's the case, then Trustly can't support `Capture`, no?

Author: Correct, this has been removed

transaction.

An example of an abbreviated `POST` request is provided below. Each individual field of the JSON document is described in the following section.
An example of an expanded `POST` request is available in the
[other features section].

When properly set up in your merchant/webshop site and the payer starts the
payment process, you need to make a POST request towards Swedbank Pay with your

> payment information. This will generate a payment object with a unique

Reviewer: ```suggestion
payment information. This will generate a `payment` resource with a unique
```

> `paymentID`. You will receive a **JavaScript source** in response.

Reviewer: Please write the name of the operation in which the JavaScript can be found. `view-payment`, is it?

```suggestion
`id` URI. You will receive a **JavaScript source** in response.
```

        {
            "method": "PATCH",
            "href": "{{ page.api_url }}/psp/trustly/payments/{{ page.payment_id }}",
            "rel": "update-payment-abort"
        },
        {
            "method": "POST",
            "href": "{{ page.api_url }}/psp/trustly/payments/{{ page.payment_id }}/sales",
            "rel": "create-sale"
        },
        {
            "method": "GET",
            "href": "https://ecom.externalintegration.payex.com/trustly/payments/sales/8f3ba6c8f4e3f6125ae6c18bec15c612747cf2c35dc5cac35d4bebc10cf7317e",
            "rel": "redirect-sale"
        },
        {
            "method": "GET",
            "href": "https://ecom.externalintegration.payex.com/trustly/core/scripts/client/px.trustly.client.js?token=8f3ba6c8f4e3f6125ae6c18bec15c612747cf2c35dc5cac35d4bebc10cf7317e&operation=sale",
            "rel": "view-sales",
            "contentType": "application/javascript"
        }
    ]
}
```

The key information in the response is the `view-sales` operation. You
will need to embed its `href` in a `<script>` element. The script will enable
loading the payment page in an `iframe` in our next step.

## Step 2: Display the Payment

> You need to embed the script source on your site to create a hosted-view in an

Reviewer: ```suggestion
You need to embed the script source on your site to create a Seamless View in an
```

    activate Merchant
    Merchant-->>-Consumer: Display all details and final price
    activate Consumer
    note left of Consumer: Open iframe ③
    Consumer->>Consumer: Input first name and last name
    Consumer->>-SwedbankPay: Confirm purchase
    activate SwedbankPay
    SwedbankPay->>-Trustly: Perform payment
    activate Trustly
    Trustly-->>-Merchant: Transaction result
    activate Merchant
    note left of Merchant: Second API request
    Merchant->>-SwedbankPay: GET <payment.id>
    activate SwedbankPay
    SwedbankPay-->>-Merchant: payment resource
    activate Merchant
    Merchant-->>-Consumer: Display result
```

{% include iterator.html prev_href="redirect" prev_title="Back: Redirect"
next_href="direct" next_title="Next: Direct" %}

[after-payment]: /payments/trustly/after-payment
[authorize]: /payments/trustly/other-features#authorizations
[callback]: /payments/trustly/other-features#callback
[cancel]: /payments/trustly/after-payment#cancellations
[capture]: /payments/trustly/capture
[create-payment]: /payments/trustly/other-features#create-payment
[fi-png]: /assets/img/flag-finland.png
[financing-consumer]: /payments/trustly/other-features#financing-consumer

> [trustly-payment-embedded-view]: /assets/screenshots/trustly/consumer-information-view.png

Author: Fixed 👍  wasn't aware snipping tool saved as .PNG instead of .png lol

---

> title: Swedbank Pay Trustly Payments Seamless view

Reviewer: ```suggestion
title: Swedbank Pay Trustly Payments – Seamless view
```

sidebar:
  navigation:
  - title: Trustly Payments
    items:
    - url: /payments/trustly
      title: Introduction
    - url: /payments/trustly/redirect
      title: Redirect
    - url: /payments/trustly/seamless-view
      title: Seamless View
    - url: /payments/trustly/after-payment
      title: After Payment
    - url: /payments/trustly/other-features
      title: Other Features
---

{% include jumbotron.html body="The Seamless View scenario represents the
                                opportunity to implement Trustly directly in
                                your webshop." %}

## Introduction

Seamless View provides an integration of the payment process directly on your
website. This solution offers a smooth shopping experience with Swedbank Pay
Payments seamlessly integrated in an `iframe` on your website. The payer
does not need to leave your webpage, since we are handling the payment in the
`iframe` on your website.

![screenshot of the Trustly payment window][trustly-payment-embedded-view]{:height="425px" width="700px"}

{% include alert-callback-url.md api_resource="trustly" %}

## Step 1: Create the payment

> A Trustly payment is a straightforward way to perform a direct-bank payment. It is followed up by posting a cancellation or reversal

Reviewer: ```suggestion
A Trustly payment is a straightforward way to perform a direct-bank payment. It can be followed up by posting a cancellation or reversal
```

transaction.

An example of an abbreviated `POST` request is provided below. Each individual field of the JSON document is described in the following section.
An example of an expanded `POST` request is available in the

> [other features section].

Reviewer: Missing link.

```suggestion
[other features section][???].
```

When properly set up in your merchant/webshop site and the payer starts the

> payment process, you need to make a POST request towards Swedbank Pay with your

Reviewer: ```suggestion
payment process, you need to make a `POST` request towards Swedbank Pay with your
```
